File: app/src/main/java/com/example/sampleproject_1/Database/ViewModelWaterReminder.java
package com.example.sampleproject_1.Database;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;


import java.util.List;

public class ViewModelWaterReminder extends AndroidViewModel {
    private RepositoryWaterReminder repository;
    private LiveData<List<EntityWaterReminder>> allNotes;


    public ViewModelWaterReminder(@NonNull Application application) {
        super(application);
        repository = new RepositoryWaterReminder(application);
        allNotes = repository.getAllUsers();
    }

    public void insert(EntityWaterReminder entityWaterReminder) {
        repository.insert(entityWaterReminder);
    }

    public LiveData<List<EntityWaterReminder>> getAllNotes() {
        return allNotes;
    }


}

File: app/src/main/java/com/example/sampleproject_1/WaterReminder/Database/ViewModelWaterReminder.kt
package com.example.sampleproject_1.WaterReminder.Database

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.launch

class ViewModelWaterReminder(application: Application) :AndroidViewModel(application) {

    /* private val repository: RepositoryWaterReminder = RepositoryWaterReminder(application)
    fun insert(entityWaterReminder: EntityWaterReminder?) {
        repository.insert(entityWaterReminder)
    }

    val allData: LiveData<List<EntityWaterReminder?>?>
        get() = repository.allData

    // private LiveData<List<EntityWaterReminder>> allNotes;
    init {
        // allNotes = repository.getAllUsers();
    }*/

    val getAllUser: LiveData<List<EntityWaterReminder>>
    private val repositoryWaterReminder: RepositoryWaterReminder

    init {
        val dataAccessObjectWaterReminder = DatabaseWaterReminder.getInstance(application)?.dataAccessObjectWaterReminder()
        repositoryWaterReminder = RepositoryWaterReminder(dataAccessObjectWaterReminder!!)
        getAllUser = repositoryWaterReminder.getAllUser
    }

    fun insert(entityWaterReminder: EntityWaterReminder) {
        viewModelScope.launch(Dispatchers.IO) {
            repositoryWaterReminder.insert(entityWaterReminder)
        }
    }

    fun deleteAllData(){
        viewModelScope.launch(Dispatchers.IO) {
            repositoryWaterReminder.deleteAllData()
        }
    }

}
